barretenberg/cpp/src/barretenberg/dsl/acir_format/serde/witness_stack.hpp

```diff
@@ -1,49 +1,141 @@
 #pragma once
 
 #include "bincode.hpp"
+#include "msgpack.hpp"
 #include "serde.hpp"
 
-namespace WitnessStack {
+namespace Witnesses {
+struct Helpers {
```

Comment:
It's rare to have a struct with just static methods, they can be free functions. if you want to group them you can use a namespace.

Reply:
That's how it started, however that doesn't seem to work with wasm. I got errors from `wasm-ld` telling me that despite `#pragma once`, I have duplicated methods: IIRC `Acir::Program::make_kvmap` appears in `acir_format_whatever.cpp.o` and `acir_format_whatever_mocks.cpp.o`, basically everywhere I included the header. I saw in https://www.fsdeveloper.com/forum/threads/wasm-question-is-wasm-ignoring-preprocessor-command-in-some-cases.457845/ that a definition of a type should be fine, but other kinds of data might not be, so I put it in a struct, like all the other stuff here. 

NB this `Helper` can be removed if we use `MSGPACK_FIELDS`. I left it for review feedback, for example maybe we want to reject extra fields, but if it's icky we can delete it.

Reply:
Removed the `Helper` in favour of `MSGPACK_FIELDS` for now.